src/Simplex/Messaging/Agent.hs

```diff
@@ -407,8 +463,8 @@ processSMPTransmission c@AgentClient {subQ} (srv, rId, cmd) = do
           msg <- decryptAndVerify rq msgBody
           let msgHash = C.sha256Hash msg
           case parseSMPMessage msg of
-            Left e -> notify $ ERR e
-            Right (SMPConfirmation senderKey) -> smpConfirmation senderKey
+            Left e -> notify c connId $ ERR e
```

Comment:
rather than changing all calls to local notify for the sake of taking it out, it would be better to just use another function where it was needed (or just its implementation - it's one line).

The whole point of having it was not to pass these parameters at call sites.

Reply:
done